### PR TITLE
Fix logs and env (FT-5052)

### PR DESCRIPTION
**Pull request by pedrocodacy**

### CODE REVIEW

File: src/main/scala/com/codacy/analysis/cli/clients/CodacyClient.scala [hunk 1]
```diff
@@ -69,7 +69,7 @@ class CodacyClient(credentials: Credentials, http: HttpHelper)(implicit context:
           logger.error(error)(s"Error posting data to endpoint $endpoint")
           Left(error.message)
         case Right(json) =>
-          logger.info(s"""Success posting batch of results to endpoint "$endpoint" """)
+          logger.info(s"""Success posting batch of ${results.size} results to endpoint "$endpoint" """)
```
Comment (rtfpessoa) [inline src/main/scala/com/codacy/analysis/cli/clients/CodacyClient.scala]: 👍

File: src/main/scala/com/codacy/analysis/cli/Main.scala [hunk 2]
```diff
@@ -50,17 +50,19 @@ class MainImpl extends CLIApp {
...
-            logger.error(err.getMessage)
-            Left(err.getMessage)
-          case Success(Left(err)) =>
-            logger.warn(err)
-            Left(err)
-          case Success(Right(_)) =>
-            logger.info("Completed upload of results to API")
-            Right(())
-        }
+        val uploadResult = if (analyse.uploadValue) {
```
Comment (rtfpessoa) [inline src/main/scala/com/codacy/analysis/cli/Main.scala]: 👍

File: bin/codacy-analysis-cli.sh [hunk 3]
```diff
@@ -29,6 +29,9 @@ run() {
   docker run \
     --rm \
     --env CODACY_CODE="$CODACY_CODE" \
+    --env CODACY_PROJECT_TOKEN="$CODACY_PROJECT_TOKEN" \
```
Comment (rtfpessoa) [inline bin/codacy-analysis-cli.sh]: 👍

File: src/main/scala/com/codacy/analysis/cli/configuration/Environment.scala [hunk 4]
```diff
@@ -60,7 +60,9 @@ class Environment(variables: Map[String, String]) {
 
   private def validate(name: String, paramType: String, param: String)(value: Option[String]): Option[String] = {
     value.ifEmpty(logger.info(s"$name not passed through $paramType `$param`")).flatMap {
-      case t if t.trim.nonEmpty => Option(t.trim)
+      case t if t.trim.nonEmpty =>
+        logger.info(s"$name found in $paramType `$param`")
```
Comment (rtfpessoa) [inline src/main/scala/com/codacy/analysis/cli/configuration/Environment.scala]: 👍

File: src/main/scala/com/codacy/analysis/cli/clients/CodacyClient.scala [hunk 5]
```diff
@@ -103,20 +103,22 @@ class CodacyClient(credentials: Credentials, http: HttpHelper)(implicit context:
     }
   }
 
-  private def validateRemoteResultsResponse(json: Json): Either[String, Unit] = {
-    parse[RemoteResultResponse]("sending results", json).map { _ =>
+  private def validateRemoteResultsResponse(json: Json, isEnd: Boolean = false): Either[String, Unit] = {
+    val action = if (isEnd) "end of results" else "sending results"
+    val message = s"Endpoint for $action replyed with an error"
```
Comment (rtfpessoa) [inline src/main/scala/com/codacy/analysis/cli/clients/CodacyClient.scala]: `replyed` -> `replied`

File: src/test/scala/com/codacy/analysis/cli/CodacyClientSpec.scala [hunk 6]
```diff
@@ -121,7 +121,7 @@ class CodacyClientSpec extends Specification with NoLanguageFeatures with Mockit
             // scalafix:on NoInfer.any
             endOfResults must beLike[Either[String, Unit]] {
               case Left(errorMsg) =>
-                errorMsg mustEqual "Error sending results: failed!"
+                errorMsg mustEqual "Error: Endpoint for end of results replyed with an error : failed!"
```
Comment (xplosunn) [inline src/test/scala/com/codacy/analysis/cli/CodacyClientSpec.scala]: Same typo: replyed -> replied

File: src/test/scala/com/codacy/analysis/cli/CodacyClientSpec.scala [hunk 7]
```diff
@@ -90,7 +90,7 @@ class CodacyClientSpec extends Specification with NoLanguageFeatures with Mockit
             // scalafix:on NoInfer.any
             remoteResults must beLike[Either[String, Unit]] {
               case Left(errorMsg) =>
-                errorMsg mustEqual "Error sending results: failed!"
+                errorMsg mustEqual "Error: Endpoint for sending results replyed with an error : failed!"
```
Comment (xplosunn) [inline src/test/scala/com/codacy/analysis/cli/CodacyClientSpec.scala]: Same typo: replyed -> replied

**Pull request merged**